test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -288,3 +267,87 @@ func TestVLANNetwork(t *testing.T) {
 	}
 	testSecondaryNetwork(t, networkTypeVLAN, pods)
 }
+
+func (data *testData) assignIP(clientset *kubernetes.Clientset) error {
+	e2eTestData := data.e2eTestData
+	namespace := e2eTestData.GetTestNamespace()
+
+	for _, testPod := range data.pods {
+		node, err := clientset.CoreV1().Nodes().Get(context.TODO(), testPod.nodeName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("error when getting the cluster Node %s: %v", testPod.nodeName, err)
+		}
+		eni, has := node.Labels["eni-id"]
+		if !has {
+			return fmt.Errorf("the label `eni-id` not found in the cluster Node: %s", testPod.nodeName)
+		}
+		var podIP net.IP
+		_, err = e2eTestData.PodWaitFor(defaultTimeout, testPod.podName, namespace, func(pod *corev1.Pod) (bool, error) {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false, nil
+			}
+			podIPs, err := data.listPodIPs(testPod)
+			if err != nil {
+				return false, err
+			}
+			ip, exists := podIPs["eth1"]
+			if !exists || ip == nil {
+				logs.Infof("IP not available for interface 'eth1' in Pod %s, retrying...", testPod.podName)
+				return false, nil
+			}
+			podIP = ip
+
+			return true, nil
+		})
+		if err := aws.AssignIPToEC2ENI(context.TODO(), eni, podIP.String()); err != nil {
+			return err
+		}
+		logs.Infof("assigned private IP address %s to interface %s", podIP, eni)
+		if err != nil {
+			return fmt.Errorf("error when waiting for the secondary IP for Pod %+v: %w", testPod, err)
+		}
+	}
+	return nil
+}
+
+func TestSRIOVNetwork(t *testing.T) {
+	e2eTestData, err := antreae2e.SetupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer antreae2e.TeardownTest(t, e2eTestData)
+
+	node0Name := antreae2e.NodeName(0)
+	node1Name := antreae2e.NodeName(1)
```

Comment:
nit: no need to define those variables, it just makes the test more verbose

you can use `nodeName: antreae2e.NodeName(0)` directly below